### PR TITLE
Resolve PE bugs

### PR DESCRIPTION
**Pull request by jordankanghm**
Resolved the following issues.

1. Events with same name but different fields are not allowed
	-Indicated in UG that event names are identity fields.

2. Description and Location cannot take brackets
	-Changed the regex for Description and Location such that they can accept all characters

3. "eaddv -> vedit -> eremovev" and "eaddv -> vedit -> elistv" does not work as intended
        -Indicated in UG that volunteer names are identity fields, hence vedit will not be able to change the names of volunteers.

4. Corrected the error message for ecreate to show a valid example

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -113,6 +113,7 @@ Parameters:
 </box>
 
 Restrictions:
+* Volunteers cannot have duplicate names(Volunteer names are unique to each volunteer).
```
Comment (brein62) [inline docs/UserGuide.md:116]: perhaps can add space between `names` and `(` so we don't get a `veryLow` documentation bug hahah

File: docs/UserGuide.md [hunk 2]
```diff
@@ -241,8 +242,10 @@ Parameters:
 Restrictions:
 * All parameters must be separated by a single space.
 * All arguments cannot be blank.
+* Events cannot have duplicate names(Event names are unique to each event).
```
Comment (brein62) [inline docs/UserGuide.md:245]: likewise here

File: docs/UserGuide.md [hunk 3]
```diff
@@ -241,8 +242,10 @@ Parameters:
...
 * The date and time formats must be exactly `DD/MM/YYYY TTTT`.
 * If the end date and time is specified, it must be the _same time_ or _after_ the start date and time of the event.
+* The location and description arguments cannot contain the `[` and `]` symbols.
```
Comment (brein62) [inline docs/UserGuide.md:248]: oo is this a limitation of the regex engine? cos i think can use `\[` and `\]` to bypass this issue

**Review by brein62 — CHANGES_REQUESTED** [review]
seems okies, but i don't think that a simple UG change is enough to just fix some of these issues leh...

for #188, can explore whether can support `[` or `]` or not first hahah

for #180 and #176, my suggestion is below: 

best to remove the name field from being edited in vedit due to name being an identity field in the Volunteer model. Likewise to be done with event name in Event model. (since a feature removal is perfectly fine if it causes bugs)

File: docs/UserGuide.md (hunk 3 above)
Comment (jordankanghm) [reply docs/UserGuide.md:248]: ya i think cannot leh. my current regex alr has `\` before the `]`

Comment (brein62) [reply docs/UserGuide.md:248]: try the test case `ecreate n/go jb shopping sd/31/12/2023 2359 r/10 shopper l/j]b m/20 shopping bags dsc/shopping]]]`

Comment (brein62) [reply docs/UserGuide.md:248]: it's possible that `]` works while `[` doesn't cos i don't see `\[` in the regex

File: src/main/java/seedu/address/model/event/Location.java [hunk 4]
```diff
@@ -15,7 +15,7 @@ public class Location {
      * The first character of the location must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
-    public static final String VALIDATION_REGEX = "[A-Za-z0-9 _.,!\"'/$]+";
+    public static final String VALIDATION_REGEX = "^[a-zA-Z0-9 !\\\"#$%&'()*+,-./:;<=>?@[\\\\]^_` ]+$";
```
Comment (brein62) [inline src/main/java/seedu/address/model/event/Location.java:18]: try this regex: ```^[a-zA-Z0-9 !\\\"#$%&'()*+,-./:;<=>?@\[\\\\]^_` ]+$```

File: docs/UserGuide.md (hunk 3 above)
Comment (jordankanghm) [reply docs/UserGuide.md:248]: wah lets go jb ah

File: src/main/java/seedu/address/model/event/Location.java (hunk 4 above)
Comment (jordankanghm) [reply src/main/java/seedu/address/model/event/Location.java]: this one also cannot leh, more buggy than the original regex actually

**Review by zacwong2151 — APPROVED** [review]
looking good

File: docs/UserGuide.md (hunk 3 above)
Comment (brein62) [reply docs/UserGuide.md:248]: yessir! anyway does it work lol

File: src/main/java/seedu/address/model/event/Location.java (hunk 4 above)
Comment (brein62) [reply src/main/java/seedu/address/model/event/Location.java]: oof. i think should just use `.` - https://stackoverflow.com/questions/2912894/how-to-match-any-character-in-regular-expression

**Review by brein62 — APPROVED** [review]
awesome!

**Comment by brein62** [comment]
but i think the regex fails the invalid test case so i think a good example of invalid location etc. could be `" "`

**Pull request closed without merging**